org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/RefactorProcessor.java

```diff
@@ -346,7 +346,7 @@ private boolean getInlineProposal(IInvocationContext context, ASTNode node, Coll
 				// Inline Local Variable
 				if (binding.getJavaElement() instanceof ILocalVariable && RefactoringAvailabilityTesterCore.isInlineTempAvailable((ILocalVariable) binding.getJavaElement())) {
 					InlineTempRefactoring refactoring= new InlineTempRefactoring((VariableDeclaration) decl);
-					if (refactoring.checkInitialConditions(new NullProgressMonitor()).isOK()) {
+					if (refactoring.checkInitialConditions(new NullProgressMonitor()).isOK() && refactoring.getReferences().length > 0) {
```

Comment:
Since `refactoring.checkInitialialConditions()` will also invoke `refactoring.getReferences()` and cache the result to a field. The second call `refactoring.getReferences()` will just return the previous result, and no extra cost.